### PR TITLE
Add anti-affinity create group call

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
The Exoscale API now exposes dedicated Anti-Affinity Group calls for managing Anti-Affinity Groups in addition to Affinity Groups: add support for those new calls in egoscale.

Tested on my side

### CODE REVIEW

**Comment by pierre-emmanuelJ** [comment]
> You're forgetting related `listAntiAffinityGroups` and `deleteAntiAffinityGroup` API calls: https://community.exoscale.com/api/compute/#antiaffinitygroups

thanks

File: affinity_groups.go [hunk 1]
```diff
@@ -156,3 +156,54 @@ type ListAffinityGroupTypesResponse struct {
...
+func (CreateAntiAffinityGroup) Response() interface{} {
+	return new(AsyncJobResult)
+}
+
+// AsyncResponse returns the struct to unmarshal the async job
+func (CreateAntiAffinityGroup) AsyncResponse() interface{} {
+	return new(AffinityGroup)
+}
+
+// ListAntiAffinityGroups represents an (anti-)affinity groups search
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAntiAffinityGroups represents an Anti-Affinity Groups search
```

File: affinity_groups.go [hunk 2]
```diff
@@ -156,3 +156,54 @@ type ListAffinityGroupTypesResponse struct {
 	Count             int                 `json:"count"`
 	AffinityGroupType []AffinityGroupType `json:"affinitygrouptype"`
 }
+
+// CreateAntiAffinityGroup represents an anti-affinity Group creation
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// CreateAntiAffinityGroup represents an Anti-Affinity Group creation.
```

File: affinity_groups.go [hunk 3]
```diff
@@ -156,3 +156,54 @@ type ListAffinityGroupTypesResponse struct {
...
+type CreateAntiAffinityGroup struct {
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	_           bool   `name:"createAntiAffinityGroup" description:"Creates an anti-affinity group"`
+}
+
+// Response returns the struct to unmarshal
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// Response returns an empty structure to unmarshal an Anti-Affinity Group creation API response into.
```

Idem for other `Reponse()` methods in this PR :wink:

File: affinity_groups.go [hunk 4]
```diff
@@ -156,3 +156,54 @@ type ListAffinityGroupTypesResponse struct {
...
+	ID               *UUID  `json:"id,omitempty" doc:"List the anti-affinity group by the ID provided"`
+	Keyword          string `json:"keyword,omitempty" doc:"List by keyword"`
+	Name             string `json:"name,omitempty" doc:"Lists anti-affinity groups by name"`
+	Page             int    `json:"page,omitempty"`
+	PageSize         int    `json:"pagesize,omitempty"`
+	VirtualMachineID *UUID  `json:"virtualmachineid,omitempty" doc:"Lists anti-affinity groups by virtual machine ID"`
+	_                bool   `name:"listAntiAffinityGroups" description:"Lists anti-affinity groups"`
+}
+
+// ListAntiAffinityGroupsResponse represents a list of (anti-)affinity groups
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAntiAffinityGroupsResponse represents a list of Anti-Affinity Groups.
```

File: affinity_groups.go (hunk 1 above)
Comment (pierre-emmanuelJ) [reply affinity_groups.go]: oh I forgot that ones

File: affinity_groups.go (hunk 3 above)
Comment (pierre-emmanuelJ) [reply affinity_groups.go]: I wanted to stay consistent with the rest of the file

Comment (falzm) [reply affinity_groups.go]: Fair enough 😏

**Comment by pierre-emmanuelJ** [comment]
I Generated Implementation of `Listable` interface and fix doc error from listApis

File: affinity_groups.go [hunk 5]
```diff
@@ -6,28 +6,28 @@ import (
 	"net/url"
 )
 
-// AffinityGroup represents an (anti-)affinity group
+// AffinityGroup represents an (Anti-)Affinity Group.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// AffinityGroup represents an Affinity Group.
```

File: affinity_groups.go [hunk 6]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
 
-// AffinityGroupType represent an affinity group type
+// AffinityGroupType represent an Affinity Group type.
 type AffinityGroupType struct {
-	Type string `json:"type,omitempty" doc:"the type of the affinity group"`
+	Type string `json:"type,omitempty" doc:"the type of the Affinity Group"`
 }
 
-// CreateAffinityGroup (Async) represents a new (anti-)affinity group
+// CreateAffinityGroup (Async) represents a new (Anti-)Affinity Group.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// CreateAffinityGroup (Async) represents a new Affinity Group.
```

File: affinity_groups.go [hunk 7]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
 }
 
-// AsyncResponse returns the struct to unmarshal the async job
+// AsyncResponse returns the struct to unmarshal the async job.
 func (CreateAffinityGroup) AsyncResponse() interface{} {
 	return new(AffinityGroup)
 }
 
-// UpdateVMAffinityGroup (Async) represents a modification of a (anti-)affinity group
+// UpdateVMAffinityGroup (Async) represents a modification of a (Anti-)Affinity Group.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// UpdateVMAffinityGroup (Async) represents a modification of an Affinity Group.
```

File: affinity_groups.go [hunk 8]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
-	_                  bool     `name:"updateVMAffinityGroup" description:"Updates the affinity/anti-affinity group associations of a virtual machine. The VM has to be stopped and restarted for the new properties to take effect."`
+	AffinityGroupIDs   []UUID   `json:"affinitygroupids,omitempty" doc:"comma separated list of Affinity Groups id that are going to be applied to the virtual machine. Should be passed only when vm is created from a zone with Basic Network support. Mutually exclusive with securitygroupnames parameter"`
+	AffinityGroupNames []string `json:"affinitygroupnames,omitempty" doc:"comma separated list of Affinity Groups names that are going to be applied to the virtual machine. Should be passed only when vm is created from a zone with Basic Network support. Mutually exclusive with securitygroupids parameter"`
+	_                  bool     `name:"updateVMAffinityGroup" description:"Updates the Affinity/Anti-Affinity Group associations of a virtual machine. The VM has to be stopped and restarted for the new properties to take effect."`
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
	_                  bool     `name:"updateVMAffinityGroup" description:"Updates the Affinity Group associations of a virtual machine. The VM has to be stopped and restarted for the new properties to take effect."`
```

File: affinity_groups.go [hunk 9]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
 }
 
-// AsyncResponse returns the struct to unmarshal the async job
+// AsyncResponse returns the struct to unmarshal the async job.
 func (UpdateVMAffinityGroup) AsyncResponse() interface{} {
 	return new(VirtualMachine)
 }
 
-// DeleteAffinityGroup (Async) represents an (anti-)affinity group to be deleted
+// DeleteAffinityGroup (Async) represents an (Anti-)Affinity Group to be deleted.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// DeleteAffinityGroup (Async) represents an Affinity Group to be deleted.
```

File: affinity_groups.go [hunk 10]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
-// AsyncResponse returns the struct to unmarshal the async job
+// AsyncResponse returns the struct to unmarshal the async job.
 func (DeleteAffinityGroup) AsyncResponse() interface{} {
 	return new(BooleanResponse)
 }
 
 //go:generate go run generate/main.go -interface=Listable ListAffinityGroups
 
-// ListAffinityGroups represents an (anti-)affinity groups search
+// ListAffinityGroups represents an (Anti-)Affinity Groups search.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAffinityGroups represents an Affinity Groups search.
```

File: affinity_groups.go [hunk 11]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
-	Type             string `json:"type,omitempty" doc:"Lists affinity groups by type"`
-	VirtualMachineID *UUID  `json:"virtualmachineid,omitempty" doc:"Lists affinity groups by virtual machine ID"`
-	_                bool   `name:"listAffinityGroups" description:"Lists affinity groups"`
+	Type             string `json:"type,omitempty" doc:"Lists Affinity Groups by type"`
+	VirtualMachineID *UUID  `json:"virtualmachineid,omitempty" doc:"Lists Affinity Groups by virtual machine ID"`
+	_                bool   `name:"listAffinityGroups" description:"Lists Affinity Groups"`
 }
 
-// ListAffinityGroupsResponse represents a list of (anti-)affinity groups
+// ListAffinityGroupsResponse represents a list of (Anti-)Affinity Groups.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAffinityGroupsResponse represents a list of Affinity Groups.
```

File: affinity_groups.go [hunk 12]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
 type ListAffinityGroupsResponse struct {
 	Count         int             `json:"count"`
 	AffinityGroup []AffinityGroup `json:"affinitygroup"`
 }
 
-// ListAffinityGroupTypes represents an (anti-)affinity groups search
+// ListAffinityGroupTypes represents an (Anti-)Affinity Groups search.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAffinityGroupTypes represents an Affinity Group types search.
```

File: affinity_groups.go [hunk 13]
```diff
@@ -44,114 +44,114 @@ func (ag AffinityGroup) Delete(ctx context.Context, client *Client) error {
...
 }
 
-// Response returns the struct to unmarshal
+// Response returns the struct to unmarshal.
 func (ListAffinityGroupTypes) Response() interface{} {
 	return new(ListAffinityGroupTypesResponse)
 }
 
-// ListAffinityGroupTypesResponse represents a list of (anti-)affinity group types
+// ListAffinityGroupTypesResponse represents a list of (Anti-)Affinity Group types.
```
Comment (falzm) [inline affinity_groups.go]:
```suggestion
// ListAffinityGroupTypesResponse represents a list of Affinity Group types.
```

File: antiaffinity_groups.go [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+// AntiAffinityGroup represents an Anti-Affinity Group.
+type AntiAffinityGroup struct {
+	Account           string `json:"account,omitempty" doc:"the account owning the Anti-Affinity Group"`
+	Description       string `json:"description,omitempty" doc:"the description of the Anti-Affinity Group"`
+	ID                *UUID  `json:"id,omitempty" doc:"the ID of the Anti-Affinity Group"`
+	Name              string `json:"name,omitempty" doc:"the name of the Anti-Affinity Group"`
+	Type              string `json:"type,omitempty" doc:"the type of the Anti-Affinity Group"`
+	VirtualMachineIDs []UUID `json:"virtualmachineIds,omitempty" doc:"virtual machine Ids associated with this affinity group"`
```
Comment (falzm) [inline antiaffinity_groups.go]:
```suggestion
	VirtualMachineIDs []UUID `json:"virtualmachineIds,omitempty" doc:"virtual machine IDs associated with this Anti-Affinity Group"`
```

**Comment by falzm** [comment]
[ch7424]

**Pull request merged**